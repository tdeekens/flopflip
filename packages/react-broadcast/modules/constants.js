export const FLAGS_CHANNEL = '@flopflip';
