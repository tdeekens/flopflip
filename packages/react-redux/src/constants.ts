export const STATE_SLICE = '@flopflip';
