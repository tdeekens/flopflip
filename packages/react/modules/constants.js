export const DEFAULT_FLAG_NAME = 'isFeatureEnabled';
export const ALL_FLAGS = '__flopflop_all_flags__';
