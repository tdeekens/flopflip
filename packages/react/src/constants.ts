export const DEFAULT_FLAG_PROP_KEY = 'isFeatureEnabled';
export const DEFAULT_FLAGS_PROP_KEY = 'featureToggles';
export const ALL_FLAGS_PROP_KEY = '@flopflip/flags';
