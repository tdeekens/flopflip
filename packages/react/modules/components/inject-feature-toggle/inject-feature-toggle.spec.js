import React from 'react';
import { mount } from 'enzyme';
import { ALL_FLAGS_PROP_KEY } from '../../constants';
import injectFeatureToggle from './inject-feature-toggle';

describe('injecting', () => {
  const TestComponent = () => <React.Fragment>Test</React.Fragment>;
  TestComponent.displayName = 'TestComponent';
  TestComponent.propTypes = {};

  const flagName = 'aFeatureToggle';
  const createTestProps = custom => ({
    [ALL_FLAGS_PROP_KEY]: { [flagName]: true },
    ...custom,
  });

  let Component;
  let wrapper;

  describe('with `propKey`', () => {
    let props;
    const propKey = 'fooFlagPropName';

    beforeEach(() => {
      props = createTestProps();

      Component = injectFeatureToggle(flagName, propKey)(TestComponent);
      wrapper = mount(<Component {...props} />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should pass the feature toggle's state as a `prop` of `propKey`", () => {
      expect(wrapper.find(TestComponent)).toHaveProp(
        propKey,
        props[ALL_FLAGS_PROP_KEY][flagName]
      );
    });
  });

  describe('without `propKey`', () => {
    let props;

    beforeEach(() => {
      props = createTestProps();

      Component = injectFeatureToggle(flagName)(TestComponent);
      wrapper = mount(<Component {...props} />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should pass the feature toggle's state as a `prop` of `isFeatureEnabled`", () => {
      expect(wrapper.find(TestComponent)).toHaveProp('isFeatureEnabled', true);
    });
  });

  describe('with non defined flagName', () => {
    let props;

    beforeEach(() => {
      const anotherFlagName = 'anotherFeatureToggle';
      props = createTestProps();

      Component = injectFeatureToggle(anotherFlagName)(TestComponent);
      wrapper = mount(<Component {...props} />);
    });

    it("should pass the feature toggle's state as `false`", () => {
      expect(wrapper.find(TestComponent)).toHaveProp('isFeatureEnabled', false);
    });
  });

  describe('with multi variate flag', () => {
    let props;

    beforeEach(() => {
      props = createTestProps({ [flagName]: 'blue' });

      Component = injectFeatureToggle(flagName)(TestComponent);
      wrapper = mount(<Component {...props} />);
    });

    it("should pass the feature toggle's state as the value", () => {
      expect(wrapper.find(TestComponent)).toHaveProp(flagName, 'blue');
    });
  });
});
