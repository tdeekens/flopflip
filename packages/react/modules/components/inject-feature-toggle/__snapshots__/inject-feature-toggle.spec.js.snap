// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`injecting with \`propKey\` should match snapshot 1`] = `
<withProps(omitProps(TestComponent))
  @flopflip/flags={
    Object {
      "aFeatureToggle": true,
    }
  }
>
  <omitProps(TestComponent)
    @flopflip/flags={
      Object {
        "aFeatureToggle": true,
      }
    }
    fooFlagPropName={true}
  >
    <TestComponent
      fooFlagPropName={true}
    >
      Test
    </TestComponent>
  </omitProps(TestComponent)>
</withProps(omitProps(TestComponent))>
`;

exports[`injecting without \`propKey\` should match snapshot 1`] = `
<withProps(omitProps(TestComponent))
  @flopflip/flags={
    Object {
      "aFeatureToggle": true,
    }
  }
>
  <omitProps(TestComponent)
    @flopflip/flags={
      Object {
        "aFeatureToggle": true,
      }
    }
    isFeatureEnabled={true}
  >
    <TestComponent
      isFeatureEnabled={true}
    >
      Test
    </TestComponent>
  </omitProps(TestComponent)>
</withProps(omitProps(TestComponent))>
`;
